standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -10800,53 +10801,89 @@ public void addNotificationEvent(NotificationEvent entry) throws MetaException {
 
   @Override
   public void cleanNotificationEvents(int olderThan) {
```

Comment:
The same improvement of deleting in batches can be applied to cleanWriteNotificationEvents also.

Reply:
Will do in separate JIRA after this new code gets some usage.